darts/anomaly_detection/anomaly_model.py

```diff
@@ -0,0 +1,214 @@
+from cProfile import label
+from darts import TimeSeries
+import numpy as np
+from typing import Union, Any, Dict, Sequence, Tuple
+from abc import ABC, abstractmethod
+from darts.datasets import AirPassengersDataset
+import pandas as pd
+import darts
+import matplotlib.pyplot as plt
+from darts.logging import raise_if, raise_if_not
+
+from darts.anomaly_detection.score import _TrainableScorer
+from darts.anomaly_detection import score
+
+from darts.models.forecasting.forecasting_model import ForecastingModel
+from darts.models.filtering.filtering_model import FilteringModel
+
+from darts.models import NaiveSeasonal, MovingAverage, NBEATSModel, KalmanFilter, GaussianProcessFilter, ARIMA, RegressionModel
+
+class _AnomalyModel(ABC):
+
+    @abstractmethod
+    def score(
+              self, series: Union[TimeSeries, Sequence[TimeSeries]]
+             ) -> Union[TimeSeries, Sequence[TimeSeries]]:
+        pass
+        # returns a scalar TimeSeries indicating presence of anomalies in series
+        
+        
+class ForecastingAnomalyModel(_AnomalyModel):
+
+   def __init__(self, model: ForecastingModel, scorer: score):
+        super().__init__()
+        # init with this model and scorer
+        if isinstance(model, ForecastingModel):
+            self.model = model 
+        else :
+            raise ValueError(
+            "Model must be a darts.models.forecasting not a {}".format(type(model))
+            )
+
+        self.scorer = scorer
+        
+   def fit(self, series, model_fit_kwargs={}, scorer_fit_kwargs={}):
```

Comment:
Don't use mutable default arguments, see: https://docs.python-guide.org/writing/gotchas/#:~:text=Python's%20default%20arguments%20are%20evaluated,to%20the%20function%20as%20well.